src/background-script/quick-and-dirty-migrations.ts

```diff
@@ -19,6 +19,58 @@ export interface Migrations {
 }
 
 export const migrations: Migrations = {
+    /*
+     * There was a bug in some annotation refactoring (due to url normalisation)
+     * that meant some annotations are created with a full url as the prefixed key
+     */
+    'normalise-all-annotation-url-keys': async ({
+        db,
+        storex,
+        normalizeUrl,
+    }) => {
+        // go through all annotation urls that beggin with http
+
+        // update their key to be normalised
+        // update their relations to be normalised (annot bookmarks, etc, )
+        // update social sharing too? yikes,
+
+        const annotationsNormalised = []
+
+        await db
+            .table('annotations')
+            .toCollection()
+            .filter((annot) => annot.url.startsWith('http'))
+            .modify((annot) => {
+                try {
+                    const oldUrl = annot.url
+                    const [url, id] = annot.url.split('#')
+                    annot.url = `${normalizeUrl(url)}#${id}`
+                    // Save the value for updating it's relations
+                    annotationsNormalised.push({ ...annot, ...{ oldUrl } })
+                } catch (e) {
+                    console.error('Error migrating old annotation', annot)
+                    console.error(e)
+                }
+            })
+
+        for (const annotation of annotationsNormalised) {
+            await db
+                .table('annotBookmarks')
+                .where('url')
+                .equals(annotation.oldUrl)
```

Comment:
Rather than performing 2*N dexie queries here for N normalized annots, you could reduce it to 2 queries using `.anyOf` instead of `.equals`. You'd also need to change the `.modify` to either re-perform the normalziation logic or restructure your `annotationsNormalized` array so that it's an object or Map indexed from the old URL to the new one.

i.e.
```
db.table('annotBookmarks').where('url').anyOf([...annotationsNormalizedUrlsMap.keys()]).modify(a => (
  a.url = annotationsNormalizedUrlsMap.get(a.url)
))
```

Reply:
Awesome `anyOf` was the Dexie op I was missing, Cheers Jon! I'll update the PR.